www/lobby/id.ts

```diff
@@ -1,16 +1,42 @@
-import { Router, Request, Response } from 'express';
+import { Router, Request, Response, NextFunction } from 'express';
+// import { nextTick } from 'process';
 import { User } from '../../lib';
 import { Lobby } from '../../lib/lobby';
+import { validateJwt, validateLobbyJwt } from '../../utils/jwt';
 
 export const lobby_id_router = Router();
 
+const joinLobby = async (req: Request, res: Response) => {
+  // check if token is valid
+  const lobbyId: string = req.body.lobbyId;
+  const lobby = await Lobby.fromId(lobbyId);
+
+  if (!lobby || lobbyId != req.params.id) return res.status(406).end('lobby token is invalid');
+
+  //add user to lobby
+  lobby?.addUser(req.body.userId);
+  res.status(200).send({...lobby.getClientResponse()});
+};
+
 // Join a lobby
 /**
  * Body Params: id, lobbyToken, refreshToken
  */
-lobby_id_router.post('/:id', async (req: Request, res: Response) => {
-  res.status(200).json('placeholder post');
-});
+lobby_id_router.post('/:id', validateJwt, async (req: Request, res: Response, next: NextFunction) => {
+  // check if user is valid
+  const userId: string = req.body.id;
+  const user = await User.fromId(userId);
+  if (!user) return res.status(404).json('invalid user').end();
```

Comment:
A better message would be `User does not exist`

Reply:
Ok, sounds good